jobs/build-triggers/tendrl-build-triggers-pr.yml

```diff
@@ -0,0 +1,245 @@
+          echo "TARGET_REPO: ${{TARGET_REPO}}"
+          # prepare unique identifier for this build (BUILD_HASH)
+          export BUILD_HASH=$(mktemp -du "build-{target_repo}-{package_prefix}{package_name}.XXXXXXXXXX")
+          echo "BUILD_HASH=${{BUILD_HASH}}" > ${{WORKSPACE}}/env.sh
+          # "reserve" build server for this job
+          echo "${{BUILD_HASH}}" >> ${{HOME}}/build_server_{project_name}_used_by
+
+          # prepare building server (if not already reserved)
+      - conditional-step:
+          condition-kind: shell
+          condition-command: "(( $(grep -v '^ *$' $HOME/build_server_{project_name}_status | wc -l) == 0))"
+          on-evaluation-failure: dont-run
+          steps:
+            - trigger-builds:
+                - project: 'tendrl-build-X-build_server-{project_name}-prepare'
+                  block: true
+                  property-file: env.sh
+
+      - shell: |
+          #!/bin/bash -xe
+          # load environment file with BUILD_HASH
+          source ${{WORKSPACE}}/env.sh
+          echo "GIT_COMMIT: ${{GIT_COMMIT}}"
+          echo "GIT_BRANCH: ${{GIT_BRANCH}}"
+          # parse PR# and head/merge variant
+          if [[ "${{ghprbPullId}}" ]]; then
+            PR_NUMBER=${{ghprbPullId}}
+            PR_VARIANT="merge"
+          else
+            PR_NUMBER=$(echo $GIT_BRANCH | awk -F / '{{ print $3 }}')
+            PR_VARIANT=$(echo $GIT_BRANCH | awk -F / '{{ print $4 }}')
+          fi
+          echo "PR_NUMBER: ${{PR_NUMBER}}"
+          echo "PR_VARIANT: ${{PR_VARIANT}}"
+          echo "ghprbPullId: ${{ghprbPullId}}"
+
+          # wait for build_server to be ready
+          until grep -q "READY" ${{HOME}}/build_server_{project_name}_status; do
+            sleep 10
+          done
+
+          # remote working directory
+          rwd="/tmp/${{BUILD_HASH}}"
+          # get build server hostname
+          build_server=$(grep -A1 '\[build_server\]' ${{HOME}}/build_server_{project_name}.hosts | tail -1)
+          # check if build_server is available.
+          ssh root@${{build_server}} "hostname"
+          # prepare remote working directory
+          ssh root@${{build_server}} "mkdir ${{rwd}}"
+
+          # clone git repository with source code
+          ssh root@${{build_server}} \
+            "git clone https://github.com/Tendrl/{package_name} ${{rwd}}/{package_name}"
+          # fetch and checkout the PR branch
+          branch="pr${{PR_NUMBER}}.${{PR_VARIANT}}"
+          ssh root@${{build_server}} \
+            "cd ${{rwd}}/{package_name}; \
+            git fetch origin refs/pull/${{PR_NUMBER}}/${{PR_VARIANT}}:${{branch}}; \
+            git checkout ${{branch}}; \
+            git status;"
+
+          # build srpm package
+          # FIXME: we should unify the usage of target update-release or gitversion
+          ssh root@${{build_server}} \
+            "cd ${{rwd}}/{package_name}; \
+            make update-release ;\
+            make gitversion ;\
```

Comment:
@TimothyAsir would it be possible to unify the Makefiles across the repos, so everywhere will be used `make gitversion` for updating the *Release* with timestamp and git hash? (now there are two make targets for the same action - `update-release` and `gitversion`, so I have to call there both commands and expect, that one of them fails.